src/plugins/gmx/gmxplugin.cpp

```diff
@@ -0,0 +1,184 @@
+/*
+ * GMX Tiled Plugin
+ * Copyright 2016, Jones Blunt <mrjonesblunt@gmail.com>
+ *
+ * This file is part of Tiled.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License as published by the Free
+ * Software Foundation; either version 2 of the License, or (at your option)
+ * any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along with
+ * this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "gmxplugin.h"
+
+#include "map.h"
+#include "tile.h"
+#include "tilelayer.h"
+#include "mapobject.h"
+#include "objectgroup.h"
+
+#include <QDir>
+#include <QFile>
+#include <QFileInfo>
+#include <QXmlStreamWriter>
+
+using namespace Tiled;
+using namespace Gmx;
+
+GmxPlugin::GmxPlugin()
+{
+}
+
+bool GmxPlugin::write(const Map *map, const QString &fileName)
+{
+    QFile file(fileName);
+    if (!file.open(QIODevice::WriteOnly | QIODevice::Text)) {
+        mError = tr("Could not open file for writing.");
+        return false;
+    }
+
+    QXmlStreamWriter stream(&file);
+
+    stream.setAutoFormatting(true);
+    stream.setAutoFormattingIndent(2);
+
+    stream.writeComment("This Document is generated by GameMaker, if you edit it by hand then you do so at your own risk!");
+
+    stream.writeStartElement("room");
+    stream.writeStartElement("caption");
+    stream.writeEndElement();
+
+    stream.writeTextElement("width", QByteArray::number(map->tileWidth() * map->width()));
```

Comment:
I'll have a closer look on Monday latest, but in the meantime, please use `QString::number` instead because all the `QXmlStreamWriter` functions take `QString`.